### DISCUSSION TITLE
Join 3 files separated one for each color channel ( LAB )

### DISCUSSION THREAD

**Discussion by fred888651** [ImageMagick/ImageMagick#2505] [Help]
Any ideas for how to join three files, separated into three files - one for each channel

Code for separating: 
magick.exe original.jpg -colorspace LAB -channel 0 -separate L_channel.JPG
magick.exe original.jpg -colorspace LAB -channel 1 -separate A_channel.JPG
magick.exe original.jpg -colorspace LAB -channel 2 -separate B_channel.JPG

Desired output after joining would be a JPG file 

Thanks!

VERSIONS:
ImageMagick-7.0.9-Q16
Windows 10 all updates
(PowerShell 5.1)

**Comment by snibgo** [comment]
Use "-combine", like this:
```
magick L_channel.JPG A_channel.JPG B_channel.JPG -combine -set colorspace Lab -colorspace sRGB out.jpg
```
Note that after combine the inputs into one image, we need to tell IM that the image is Lab, then convert it to sRGB.

However, I suggest you do not use JPEG as an intermediate format, because JPEG compression is lossy. PNG is lossless but slow. TIFF is usually lossless and fairly fast. Unless you need to keep those files, I suggest you use MIFF which is an internal IM format.

**Comment by fred888651** [comment]
Thanks again! You're a hero! 

Input JPG is saved from Photoshop in 100% quality so I think that means no loss?

But I understand what you mean, in order to use some other intermediate format you just change the file suffix from .JPG to .PNG / MIFF / WHATEVER-FORMAT ?

I have a photoshoot now but will test this later - and I reply back with results!

Thanks again!

**Comment by snibgo** [comment]
I don't use Photoshop, and I don't know if it writes JPG losslessly.

But when IM write a JPG, it always writes lossily. In other words, it always changes pixel values. (Strictly speaking, it isn't guaranteed to be lossy. But it almost always is.)

Yes, just use a different file extension: .miff or .png or whatever.

**Reply by fred888651** [reply]
Ok! Ill try that later to use PNG or IM - seems like the right thing to store temp files in lossless format!

Your answers helped me to produce this PowerShell script that takes all files in a given folder
Checks if it is sRGB
Separates them into 3 files
Add sharpness to the L channel
Join and export the LAB sharpened file

Tested with 6 files, where 2 was non-sRGB (EXIF tag missing)

Seems to work great

At least on my system...but as we all know...things happen then executing on some other machine :)

But for what its worth - here is the PowerShell Script I wrote today 
Use it at your own risk - make a temp test folder etc etc :)

https://cloud.comhem.se/s/13203b3b304d13a4b12b0439ed61d53a392

**Comment by snibgo** [comment]
Good stuff.

The parameters to "-sharpen" are "radius" and "sigma". See https://www.imagemagick.org/script/command-line-options.php#sharpen

You can use a single magick command to separate into L*a*b* channels, sharpen the L* channel, combine, and convert back to sRGB, so no need for intermediate files.

**Reply by fred888651** [reply]
Great! Thanks again!

Its a little slow now as I call external command several times, about 19 seconds per file.
Thats real slow.

But its ok for my needs. Im a photographer and gonna use it for a small number of files - 6-10 or so. 

So its ok. 

Maybe I look into how to do all in a imagemagick one-liner in the future. That would really speed things up I guess.

Thanks again Snigbo for your help! Much obliged!
